public/react-services/action-agents.js

```diff
@@ -10,176 +10,219 @@
  * Find more information about this on the LICENSE file.
  */
 import { WzRequest } from './wz-request';
-import { getToasts }  from '../kibana-services';
+import { getToasts } from '../kibana-services';
+import { UI_LOGGER_LEVELS } from '../../common/constants';
+import { UI_ERROR_SEVERITIES } from './error-orchestrator/types';
+import { getErrorOrchestrator } from './common-services';
 
 export class ActionAgents {
   static showToast = (color, title, text, time) => {
     getToasts().add({
       color: color,
       title: title,
       text: text,
-      toastLifeTimeMs: time
+      toastLifeTimeMs: time,
     });
   };
 
   /**
    * Upgrade unique agent to the latest version avaible.
    * @param {Number} agentId
    */
-  static upgradeAgent(agentId) {
-    WzRequest.apiReq('PUT', `/agents/${agentId}/upgrade`, {
-      force: 1
-    })
-      .then(() => {
-        console.log('Upgrading');
-      })
-      .catch(error => {
-        error !== 'Wazuh API error: 3021 - Timeout executing API request'
-          ? this.showToast('danger', 'Error upgrading agent', error, 5000)
-          : false;
+  static async upgradeAgent(agentId) {
+    try {
+      await WzRequest.apiReq('PUT', `/agents/${agentId}/upgrade`, {
+        force: 1,
       });
-    this.showToast('success', 'Upgrading agent...', '', 5000);
+      console.log('Upgrading');
+      this.showToast('success', 'Upgrading agent...', '', 5000);
+    } catch (error) {
+      const options = {
+        context: `${ActionAgents.name}.upgradeAgent`,
+        level: UI_LOGGER_LEVELS.ERROR,
+        severity: UI_ERROR_SEVERITIES.BUSINESS,
+        store: true,
+        error: {
+          error: error,
+          message: error.message || error,
+          title: `Error upgrading the agent`,
+        },
+      };
+      getErrorOrchestrator().handleError(options);
+    }
   }
 
   /**
    * Upgrade list of agents to the latest version avaible.
    * @param {Array} selectedItems
    */
-  static upgradeAgents(selectedItems) {
+  static async upgradeAgents(selectedItems) {
     for (let item of selectedItems.filter(
-      item => item.outdated && item.status !== 'Disconnected'
+      (item) => item.outdated && item.status !== 'Disconnected'
     )) {
-      WzRequest.apiReq('PUT', `/agents/${item.id}/upgrade`, '1')
-        .then(() => {})
-        .catch(error => {});
+      try {
+        WzRequest.apiReq('PUT', `/agents/${item.id}/upgrade`, '1');
```

Comment:
**nitpick**: The `await` keyword is missing in the asynchronous method to wait for the response.

I am not sure if we need the `1` as the `body` parameter.